packages/zoe/src/issuerStorage.js

```diff
@@ -0,0 +1,153 @@
+// @ts-check
+
+import { makeWeakStore as makeNonVOWeakStore } from '@agoric/store';
+import { assert } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+
+import { arrayToObj } from './objArrayConversion';
+import { cleanKeywords } from './cleanProposal';
+
+/**
+ *  Make the Issuer Storage. If `exportedIssuerStorage` is defined,
+ *  then use it as the starting state.
+ *
+ * @param {ExportedIssuerStorage=} exportedIssuerStorage
+ */
+export const makeIssuerStorage = exportedIssuerStorage => {
+  /** @type {WeakStore<Brand,IssuerRecord>} */
+  const brandToIssuerRecord = makeNonVOWeakStore('brand');
+  /** @type {WeakStore<Issuer,IssuerRecord>} */
+  const issuerToIssuerRecord = makeNonVOWeakStore('issuer');
+
+  /**
+   * If we already know the entire issuer record, such as for a
+   * ZCFMint issuer, or for an issuer that Zoe has told us about,
+   * store the issuerRecord directly.
+   *
+   * @param {IssuerRecord} issuerRecord
+   * @returns {IssuerRecord}
+   */
+  const storeIssuerRecord = issuerRecord => {
+    const { brand, issuer } = issuerRecord;
+    if (issuerToIssuerRecord.has(issuer)) {
+      return issuerToIssuerRecord.get(issuer);
+    }
```

Comment:
>  If, for example, the new record has a different brand, you'll return the record according to the issuer rather than the brand.

The new record can't have a different brand than the brand that matches the issuer according to mutual agreement. We use `storeIssuerRecord` in 4 cases: 1) ZoeMint records, 2) ZCFMint records, 3) within `storeIssuer`, and 4) storing an issuer record in ZCF that we obtained from Zoe that originally went through `storeIssuer`. In both ZoeMints and ZCFMints we create the issuers and brands ourselves using ERTP directly, so we know they are not malicious and the issuers and brands match. In the last two cases, we go through the brand-issuer-match check in `storeIssuer`. 

The reason why we need the `has` check in lines 32-34 is for the 4th case above, in which we store an issuer record in ZCF that we obtained from Zoe. `init` errors if the key is already present, and because an issuer can be used more than once in the same contract, we need to make sure we aren't trying to `init` twice. If it's already present, we do not need to add it again in ZCF.

I just tried commenting out lines 32-34 and I get 16 tests failed. Could it have been possible that you didn't `build`?

I'll admit that this could be more obvious rather than requiring an analysis, but I think it's sound as is. I can copy this write-up into a comment if that'd be helpful.

Reply:
Added a comment that is a cleaned-up version of ^

Reply:
> I just tried commenting out lines 32-34 and I get 16 tests failed. Could it have been possible that you didn't build?

Ah. yes.

Reply:
I like the new comment. Resolving.